kuwala/core/canvas/src/components/Modals/BlockCatalog/TransformationCatalog/TransformationCatalogModal.js

```diff
@@ -0,0 +1,129 @@
+import React, {Fragment, useEffect, useState} from "react";
+import {useStoreActions} from "easy-peasy";
+import {
+    getAllTransformationCatalogCategories,
+    getAllItemsInCategory
+} from "../../../../api/TransformationCatalog";
+import "./transformation-example-table.css";
+import TransformationBlockDTO from "../../../../data/dto/TransformationBlockDTO";
+import TransformationCatalogDTO from "../../../../data/dto/TransformationCatalogDTO";
+import {v4} from "uuid";
+import TransformationCatalogFooter from "./TransformationCatalogFooter";
+import TransformationCatalogSelector from "./TransformationCatalogSelector";
+import TransformationCatalogBody from "./TransformationCatalogBody";
+
+export default () => {
+    const { toggleBlockCatalogModal } = useStoreActions(actions => actions.common);
+    const { convertTransformationBlockIntoElement, addTransformationBlock } = useStoreActions(actions => actions.canvas);
+    const [selectedTransformationIndex, setSelectedTransformationIndex] = useState(null);
+    const [catalogCategories, setCatalogCategories] = useState([]);
+    const [catalogOptions, setCatalogOptions] = useState([]);
+    const [selectedCatalogOption, setSelectedCatalogOption] = useState(null);
+
+    useEffect(() => {
+        initTransformationCatalogs().then(null);
+    }, [])
+
+    useEffect(()=> {
+        if(catalogCategories.length) {
+            const category = catalogCategories[selectedTransformationIndex]
+            fetchCatalogBodyItems(category).then(null);
+        }
+    }, [selectedTransformationIndex])
```

Comment:
Also here you should add `catalogCategories`.

Reply:
And `fetchCatalogBodyItems` has to be either declared inside `useEffect` or with `useCallback` as mentioned in the last review.